File: 190-reverse-bits/reverse-bits.cpp
#include <cstdint>
#include <string>
#include <algorithm> // For reverse

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        // Convert the integer to a binary string representation
        std::string binaryStr = std::bitset<32>(n).to_string();

        // Reverse the binary string
        std::reverse(binaryStr.begin(), binaryStr.end());

        // Convert the reversed binary string back to an integer
        uint32_t result = std::bitset<32>(binaryStr).to_ulong();

        return result;
    }
};

File: 190-reverse-bits/reverse-bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        bin_num=format(n,'032b')
        # print(bin_num)
        bin_str=str(bin_num)
        # print(bin_str)
        # print(type(bin_str))
        new_str=""
        for char in reversed(bin_str):
            new_str=new_str+char
        # print(f"the new str is reversed is {new_str}")
        decimal_num=int(new_str,2)
        # print(decimal_num)
        return decimal_num

        